src/main/java/org/seqdoop/hadoop_bam/util/VCFFileMerger.java

```diff
@@ -0,0 +1,127 @@
+package org.seqdoop.hadoop_bam.util;
+
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.samtools.util.BlockCompressedOutputStream;
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.tribble.util.TabixUtils;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder;
+import htsjdk.variant.vcf.VCFHeader;
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.zip.GZIPOutputStream;
+import org.seqdoop.hadoop_bam.KeyIgnoringVCFOutputFormat;
+import org.seqdoop.hadoop_bam.VCFFormat;
+
+import static org.seqdoop.hadoop_bam.util.NIOFileUtil.asPath;
+import static org.seqdoop.hadoop_bam.util.NIOFileUtil.deleteRecursive;
+import static org.seqdoop.hadoop_bam.util.NIOFileUtil.getFilesMatching;
+import static org.seqdoop.hadoop_bam.util.NIOFileUtil.mergeInto;
+
+/**
+ * Merges headerless VCF files produced by {@link KeyIgnoringVCFOutputFormat}
+ * into a single file.
+ */
+public class VCFFileMerger {
+  /**
+   * Merge part file shards produced by {@link KeyIgnoringVCFOutputFormat} into a
+   * single file with the given header.
+   * @param partDirectory the directory containing part files
+   * @param outputFile the file to write the merged file to
+   * @param header the header for the merged file
+   * @throws IOException
+   */
+  public static void mergeParts(final String partDirectory, final String outputFile,
```

Comment:
This fails with .bcf files (even a legitimate .bcf that htsjdk can handle, and that passes the testRoundTrip test, fails the testRoundTripWithMerge test). I'm guessing its not a quick fix though, so we might want to make a separate ticket, and for now just reject them here and document the restriction.